extensions/BOP/twitch.js

```diff
@@ -0,0 +1,324 @@
+            blockType: Scratch.BlockType.COMMAND,
+            arguments: {
+              image: {
+                type: Scratch.ArgumentType.STRING,
+                defaultValue: "https://tekinical.github.io/twitchURI.png",
+              },
+              width: {
+                type: Scratch.ArgumentType.STRING,
+                defaultValue: "50",
+              },
+              height: {
+                type: Scratch.ArgumentType.STRING,
+                defaultValue: "50",
+              },
+              position: {
+                type: Scratch.ArgumentType.STRING,
+                menu: "position",
+              },
+              bottom: {
+                type: Scratch.ArgumentType.BOOLEAN,
+                defaultValue: false,
+              },
+              top: {
+                type: Scratch.ArgumentType.BOOLEAN,
+                defaultValue: false,
+              },
+            },
+          },
+          {
+            opcode: "closetext",
+            text: "close text",
+            blockType: Scratch.BlockType.COMMAND,
+          },
+          {
+            opcode: "closeimage",
+            text: "close image",
+            blockType: Scratch.BlockType.COMMAND,
+          },
+          {
+            blockType: Scratch.BlockType.LABEL,
+            text: "Utilities",
+          },
+          {
+            opcode: "true",
+            text: "true",
+            blockType: Scratch.BlockType.BOOLEAN,
+          },
+          {
+            opcode: "false",
+            text: "false",
+            blockType: Scratch.BlockType.BOOLEAN,
+          },
+        ],
+        menus: {
+          font: {
+            acceptReporters: false,
+            items: "getFonts",
+          },
+          position: {
+            acceptReporters: false,
+            items: "getPositions",
+          },
+        },
+      };
+    }
+
+    setupchat(args) {
+      if (chat == null) {
+        var Link =
+          "https://nightdev.com/hosted/obschat/?theme=bttv_dark&channel=" +
+          Scratch.Cast.toString(args.username) +
+          "&fade=" +
+          Scratch.Cast.toNumber(args.fade) +
+          "&bot_activity=" +
+          Scratch.Cast.toString(args.bots) +
+          "&prevent_clipping=" +
+          Scratch.Cast.toString(args.prevent);
+        chat = document.createElement("iframe");
+        chat.style.width = args.width;
+        chat.style.height = args.height;
+        chat.style.border = "none";
+        chat.style.position = args.position;
+        chat.setAttribute("allowtransparency", "true");
+        chat.setAttribute("src", Link);
+        Scratch.renderer.addOverlay(chat, "manual");
+      } else {
+        alert("Use close chat to close the chat.");
+      }
+    }
+
+    closechat() {
+      if (chat == null) {
+        alert("Chat hasn't been created yet!");
```

Comment:
I suggest you remove these alerts: the only people who would find this useful are people using the extension with the 
dev tools open, and no TurboWarp user is conditioned to open the dev tools to check if an extension is working.

Reply:
I realized it was bad when I made the example project "Twitch Chat Viewer", it gave me an alert because I put when green flag clicked close chat.

Reply:
Changed every close____ block to check if chat != null then close it!